phoenix-core/src/main/java/org/apache/phoenix/iterate/RegionScannerFactory.java

```diff
@@ -240,6 +245,9 @@ public boolean nextRaw(List<Cell> result) throws IOException {
                     return next;
                 }
             }
+            if (extraLimit >= 0 && --extraLimit == 0) {
```

Comment:
Should not this check be at the end of this method just before it returns?

Reply:
It does not matter as long as this is after the local index extra where filtering. I wanted to isolate that to local indexing only, to not affect the other code paths.

Reply:
Oh. You are right. We're not building the right tuple in this case if we have a tuple projector.
In fact this caused PointInTimeScanQueryIT to fail.